expression/src/main/java/io/smallrye/common/expression/Expression.java

```diff
@@ -485,6 +491,14 @@ private static Node parseString(Itr itr, final boolean allowExpr, final boolean
                             }
                             //throw Assert.unreachableCode();
                         }
+                        case '\\': {
+                            if (flags.contains(Flag.ESCAPES)) {
+                                list.add(LiteralNode.DOLLAR);
+                                itr.prev();
+                                start = itr.getNextIdx();
+                                continue;
+                            }
+                        }
```

Comment:
Why is this case needed? Isn't this handled by the `default` below?

Reply:
Not without your suggested change here: https://github.com/smallrye/smallrye-common/pull/344#discussion_r1783112429 :)

I didn't want to change it based on some of our previous conversations, but we can remove it if we change the other case.